### PR TITLE
Fix windows script return code

### PR DESCRIPTION
**Pull request by ljacomet**
There was a path where ERRORLEVEL was not set but the script failed.
This change makes sure that we return a non zero code in that case.

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test this

File: build-logic/jvm/src/main/kotlin/gradlebuild.launchable-jar.gradle.kts [hunk 1]
```diff
@@ -46,6 +46,9 @@ tasks.jar.configure {
 val startScripts = tasks.register<GradleStartScriptGenerator>("startScripts") {
     startScriptsDir.set(layout.buildDirectory.dir("startScripts"))
     launcherJar.from(tasks.jar)
+    // The trick below is to use the templates from the current code instead of the wrapper. It does not cover the case where the generation logic is updated though.
+    unixScriptTemplate.from(layout.projectDirectory.file("../plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt"))
```
Comment (blindpirate) [inline build-logic/jvm/src/main/kotlin/gradlebuild.launchable-jar.gradle.kts:50]: `../plugins` seems a bit weird. Shall we use `project(":plugins").layout.projectDirectory.file(...)`?

Comment (ljacomet) [reply build-logic/jvm/src/main/kotlin/gradlebuild.launchable-jar.gradle.kts:50]: I wonder, as it means reaching out into another project though.

**Comment by ljacomet** [comment]
@bot-gradle test this

File: build-logic/jvm/src/main/kotlin/gradlebuild.launchable-jar.gradle.kts (hunk 1 above)
Comment (eskatos) [reply build-logic/jvm/src/main/kotlin/gradlebuild.launchable-jar.gradle.kts:50]: If `layout` provided `rootDirectory` we could more naturally go from there.
+1 to keep it as is and not reach across projects. We're reaching across builds anyway :roll_eyes:

**Review by eskatos — APPROVED** [review]

**Review by blindpirate — APPROVED** [review]

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
There is a known flaky issue. @bot-gradle test and merge

**Pull request merged**